site/content/en/docs/Installation/upgrading.md

```diff
@@ -0,0 +1,116 @@
+
+### Installation with Helm
+
+Helm features capabilities for upgrading to newer versions without having to delete the older version. For details on
+how to use Helm for upgrades, see the [helm upgrade](https://helm.sh/docs/helm/#helm-upgrade) documentation.
+
+Given the above, the steps for upgrade are simpler:
+
+1. Start your maintenance window.
+1. Run `helm upgrade` with the appropriate arguments, such a `--version`, for your specific upgrade
+1. Run any other tests to ensure the Agones installation is working as expected.
+1. Close your maintenance window.
+7. Congratulations - you have now upgraded to a new version of Agones! 👍
+
+## Upgrading Kubernetes
+
+The following are strategies for safely upgrading the underlying Kubernetes cluster from one version to another.
+They may require adjustment to your particular game architecture but should provide a solid foundation for updating your cluster safely.
+
+The recommended approach is to use [multiple clusters](#multiple-clusters-1), such that the upgrade can be tested
+gradually with production load and easily rolled back if need arises.
+
+### Multiple Clusters
+
+This process is very similar to the [Agones: Multiple Cluster](#multiple-clusters) approach above.
+
+Essentially, we want to do a [blue/green deployment](https://martinfowler.com/bliki/BlueGreenDeployment.html) to the new 
+version of Kubernetes.  This means that we can slowly migrate our game servers from the older version's cluster (blue) 
+to the new version's cluster (green), and ensure nothing surprising happens during this process.
+
+This also allows easy rollback to the previous infrastructure that we already know to be working in production,
+with minimal interruptions to player experience.
+
+The following are steps to implement this:
+
+1. Create a new cluster of the same size or smaller as the current cluster, with the new version of Kubernetes
+2. Install the same version of Agones on the new cluster, as you have on the previous cluster.
+3. Deploy the same set of Fleets and/or GameServers from the old cluster into the new cluster.
+4. With your matchmaker, start sending a small percentage of your matched players' game sessions to the new cluster.
+5. Assuming everything is working successfully on the new cluster, slowly increase the percentage of matched sessions to the new cluster, until you reach 100%.
+6. Once you are comfortable with the stability of the new cluster with the new Kubernetes version, shut down the old cluster.
+7. Congratulations - you have now upgraded to a new version of Kubernetes! 👍
+
+### Single Cluster
+
+If you are upgrading a single cluster, we recommend creating a maintenance window, in which your game goes offline
+for the period of your upgrade, as there will be a short period in which Agones will be non-responsive during the node
+upgrades.
+
+1. Start your maintenance window.
+1. Scale your Fleets down to 0 and/or delete your GameServers.
```

Comment:
can you explain why? will node upgrades kill allocated game servers? is it possible to set PDBs and/or graceful termination periods on them automatically to make node upgrades work without disruption?

Reply:
Added some words to the docs - but the TL;DR is basically to prevent #398 .

We can mitigate this with separate nodepools for the controller, but for this guide, I figured it was best to write the simplest version I could, with strategies I knew would work for all circumstances. I figured we could then iterate out to more sophisticated upgrade paths as needed. 

The other option is to split this into a "separate controller node group" section, and a "everything on the same node group" section, with seperate upgrade strategies, but wasn't sure if that was too much for this instance.

I err'd towards a path that I knew would work for all use cases, since I figured stable was best for 1.0.

Reply:
> is it possible to set PDBs and/or graceful termination periods on them automatically to make node upgrades work without disruption

On this note - it's _possible_ (probably more graceful termination, we don't have control over which Pods get terminated with PDBs AFAIK?), but I don't think it's been thoroughly tested as a strategy, so I didn't want to document it at this stage. Just trying to be safe, essentially.

Reply:
That sounds reasonable. I agree that we can expand on the instructions over time.